subsys/shell/Kconfig.template.shell_log_queue_size

```diff
@@ -0,0 +1,10 @@
+config $(module)_LOG_MESSAGE_QUEUE_SIZE
+	int "Log message queue size"
+	default $(default-size)
+	help
+	  Logger threads enqueues log messages which are then processed by shell
+	  thread. Too small queue may lead to logger thread being blocked
+	  (see $(module)_LOG_MESSAGE_QUEUE_TIMEOUT). Too big queue on relatively
+	  slow shell transport may lead to situation where logs are dropped
```

Comment:
> Too big queue on relatively
	  slow shell transport may lead to situation where logs are dropped
	  because all log messages are enqueued.

I'm not sure that "too big a queue" would lead to a case of logs being dropped. Likewise, I don't think that logs would be dropped "because all log messages are enqueued.", but because the queue is full. (Then again, the bigger the queue, the lower probability that it becomes full.)

Reply:
messages are allocated from the pool in the logger subsystem. If queue in one of the backends is long all allocated messages may stuck in that queue and pool may become empty. In that case logs will be dropped because logger subsystem cannot allocate new messages. If we intent to increase queue size we must take care pool size as well.